tests/unit/parser/test_manifest.py

```diff
@@ -91,3 +112,89 @@ def test_partial_parse_safe_update_project_parser_files_partially(
         assert "full_reparse_reason" in exc_info
         assert "KeyError: 'Whoopsie!'" == exc_info["exception"]
         assert isinstance(exc_info["code"], str) or isinstance(exc_info["code"], type(None))
+
+
+class TestGetFullManifest:
+    def set_required_mocks(
+        self, mocker: MockerFixture, manifest: Manifest, mock_adapter: MagicMock
+    ):
+        mocker.patch("dbt.parser.manifest.get_adapter").return_value = mock_adapter
```

Comment:
Is it possible to do the patch inside the fixture? So users of those fixtures can just request the fixture then the patch is done?

Reply:
I tried doing so but, as noted in the commit message of https://github.com/dbt-labs/dbt-core/pull/10147/commits/791ec1d64f569a59d817a9d5e0fb4b95a4b291cf, the mocks stopped propagating when I did so. I can take another stab at it, but can't make any promises. It definitely would be nice if we could do it as a fixture though.

Reply:
Converting it to a fixture does seem to work now. Not sure why it didn't before and why it is now 🤷

Reply:
I wonder whether it was not imported properly before